### PR TITLE
lld v13.0.1 (again)

### PR DESCRIPTION
**Pull request by h-vetinari**
Rebased #33 to keep CI results for reference if we still need to fix something.

Closes #33
Closes #32
Closes #31 
Closes #30 
Closes #29

Closes https://github.com/conda-forge/staged-recipes/pull/18004

### CODE REVIEW

**Comment by h-vetinari** [comment]
OK, this is still failing despite the llvm build having `LLVM_ENABLE_LIBXAR=OFF` as well. I went and raised an issue: https://github.com/llvm/llvm-project/issues/54001

**Comment by h-vetinari** [comment]
Woohoo, a passing osx build 🥳 

Now that the setting has taken effect in llvmdev, let's see if we can avoid it here. Together with the also-finally-passing https://github.com/conda-forge/libcxx-feedstock/pull/83, this should be bringing up the last relevant bits and pieces for LLVM 13 in conda-forge 🚀

**Comment by h-vetinari** [comment]
Ping @conda-forge/lld

This is green and ready :)

**Comment by h-vetinari** [comment]
Gentle ping @isuruf

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ requirements:
     - llvmdev =={{ version }}
     - llvm
     - libxml2  # [win]
-    - zlib     # [win]
+    - zlib     # [not osx]
```
Comment (isuruf) [inline recipe/meta.yaml:30]:
```suggestion
    - zlib     # [win]
```

Comment (h-vetinari) [reply recipe/meta.yaml:30]: Happy to try again, but there [were](https://github.com/conda-forge/lld-feedstock/pull/29#issuecomment-933035120) errors without zlib:
> Missing `libz`:
> 
> ```
> [100%] Linking CXX executable ../../bin/lld
> [...]/bin/ld: warning: libz.so.1, needed by [...]/lib/libLLVM-13.so, not found (try using -rpath or -rpath-link)
> [...]/bin/ld: [...]/lib/libLLVM-13.so: undefined reference to `compress2'
> [...]/bin/ld: [...]/lib/libLLVM-13.so: undefined reference to `crc32'
> [...]/bin/ld: [...]/lib/libLLVM-13.so: undefined reference to `compressBound@ZLIB_1.2.0'
> [...]/bin/ld: [...]/lib/libLLVM-13.so: undefined reference to `uncompress'
> ```

**Comment by h-vetinari** [comment]
OK cool, looks like this worked! :)

Let me know if you want me to clean up the commit message / history, otherwise this is ready.

**Comment by isuruf** [comment]
Thanks

**Pull request merged**